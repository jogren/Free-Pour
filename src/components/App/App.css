@import url('https://fonts.googleapis.com/css?family=Libre+Baskerville:400,700&display=swap');


* {
  margin: 0;
  font-family: 'Libre Baskerville', serif;
  box-sizing: border-box;
}

.blur-filter {
  -webkit-filter: blur(2px);
  filter: blur(2px);
  z-index: 1;
}

.header-and-nav {
  background: linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5) ), url('https://image.shutterstock.com/image-vector/vector-linear-illustration-bottles-glasses-260nw-667294063.jpg');
}
