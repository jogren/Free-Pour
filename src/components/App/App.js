import React, { Component } from 'react';
import { Route, Redirect } from 'react-router-dom'
import Header from '../Header/Header';
import Nav from '../Nav/Nav';
import DrinkContainer from '../DrinkContainer/DrinkContainer';
import DrinkDetails from '../DrinkDetails/DrinkDetails';
import GameBoard from '../GameBoard/GameBoard';
import { hideSelectedDrink, setCurrentCocktails } from '../../actions';
import { fetchPopularCocktails, fetchCocktailsByGenre } from '../../apiCalls/apiCalls';
import { connect } from 'react-redux';
import './App.css';

export class App extends Component {

  componentDidMount = async () => {
    try {
      const popularCocktails = await fetchPopularCocktails();
      this.props.setCurrentCocktails(popularCocktails)
    } catch(error) {
      throw new Error(error.message)
    }
  }

  getCocktailsByGenre = async (type) => {
    if(type === 'popular') {
      try {
        const popularCocktails = await fetchPopularCocktails();
        this.props.setCurrentCocktails(popularCocktails)
      } catch (error) {
        throw new Error(error.message)
      }
    } else {
      try {
        const cocktailsByGenre = await fetchCocktailsByGenre(type)
        this.props.setCurrentCocktails(cocktailsByGenre);
      } catch(error) {
        throw new Error(error.message)
      }
    }
  }

  render() {
    const { selectedDrink, hideSelectedDrink, currentCocktails, favoriteCocktails } = this.props
    let blur = selectedDrink.name ? 'blur-filter' : ''
    return (
      <main className="App_main">
        <Route exact path="/" render={() =>
        <section>
          {selectedDrink.name && (<DrinkDetails selectedDrink={selectedDrink} hideSelectedDrink={hideSelectedDrink} />) }
          <div className={blur}>
            <Header/>
            <Nav getCocktailsByGenre={this.getCocktailsByGenre}/>
            <DrinkContainer drinks={currentCocktails}/>
          </div>
        </section>
        } />
        <Route path="/game-play" render={() =>
          <section>
            <Header />
            <GameBoard favoriteCocktails={favoriteCocktails}/>
          </section>
        }/>
      </main>
    );
  }
}

export const mapStateToProps = ({ selectedDrink, currentCocktails, favoriteCocktails }) => ({
  selectedDrink,
  currentCocktails,
  favoriteCocktails
});

const mapDispatchToProps = dispatch => ({
  hideSelectedDrink: () => dispatch(hideSelectedDrink()),
  setCurrentCocktails: cocktails => dispatch(setCurrentCocktails(cocktails))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
