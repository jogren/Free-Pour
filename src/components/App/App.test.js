import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App';

describe('App', () => {
  let wrapper;
  const mockHideSelectedDrink = jest.fn();
  const mockSetCurrentCocktails = jest.fn();

  const mockSelectedDrink = { 
    name: 'Mojito', 
    glass: 'Highball glass', 
    instructions: 'Muddle mint leaves...', 
    image: 'image.png', 
    ingredients: [
      {
        guessed: false,
        ingredient: 'Light rum',
        measure: '2-3 oz' 
      },
      {
        guessed: false,
        ingredient: 'mint',
        measure: 'leaves'
      },
      {
        guessed: false,
        ingredient: 'sugar',
        measure: '2tbsp'
      }
    ]
  }

  const mockCurrentCocktails = [
    {
      strDrink: 'Mojito',
      strDrinkThumb: 'image.png'
    },
    {
      strDrink: 'Moscow Mule',
      strDrinkThumb: 'image2.png'
    }
  ]

  const mockFavoriteCocktails = [
    {
      name: 'Moscow Mules',
      glass: 'Copper Mug',
      instructions: 'Combine ingredients...',
      image: 'image2.png',
      ingredients: [
        {
          guessed: false,
          ingredient: 'Vodka',
          measure: '2-3 oz'
        },
        {
          guessed: false,
          ingredient: 'Ginger Ale',
          measure: '5oz'
        },
        {
          guessed: false,
          ingredient: 'Lime Juice',
          measure: '2tbsp'
        }
      ]
    }
  ]

  beforeEach(() => {
    wrapper = shallow(<App 
      selectedDrink={mockSelectedDrink}
      currentCocktails={mockCurrentCocktails}
      favoriteCocktails={mockFavoriteCocktails}
      hideSelectedDrink={mockHideSelectedDrink}
      setCurrentCocktails={mockSetCurrentCocktails}
    />);
  });

  it('should take a snapshot when passed the correct data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should return an object with selectedDrink, currentCocktails, favoriteCocktails', () => {
    const mockState = {
      selectedDrink: { id: 1, name: 'Moscow Mules' },
      currentCocktails: [
        { 
          id: 1, 
          name: 'Moscow Mule'
        },
        {
          id: 2,
          name: 'Mojito'
        }
      ],
      favoriteCocktails: [
        {
          id: 1,
          name: 'Moscow Mule'
        }
      ]
    };
    const expected = {
      selectedDrink: { id: 1, name: 'Moscow Mules' },
      currentCocktails: [
        {
          id: 1,
          name: 'Moscow Mule'
        },
        {
          id: 2,
          name: 'Mojito'
        }
      ],
      favoriteCocktails: [
        {
          id: 1,
          name: 'Moscow Mule'
        }
      ]
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
})
