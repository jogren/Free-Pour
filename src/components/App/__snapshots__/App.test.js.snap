// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should take a snapshot when passed the correct data 1`] = `
<main
  className="App_main"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/game-play"
    render={[Function]}
  />
</main>
`;
